src/select.coffee

```diff
@@ -140,16 +153,39 @@ Ember.AddeparMixins.ResizeHandlerMixin, Ember.Widgets.KeyboardHelper,
       @get('originalItemViewClass')
   .property 'showTooltip'
 
+  ###*
+  * Function to be called when we want to open the dropdown, i.e. we have to
+  * clean up the query, render the dropdown...
+  * @function
+  ###
+  openDropdown: ->
+    @set 'showDropdown', yes
+    # Clear the query content when users open the dropdown so that users can
+    # start typing their new query right away.
+    @set 'query', ''
+
+    @updateDropdownLayout()
+
+    # Set the focus to the search field so that users can start typing their
+    # query right away.
+    @_setSearchFieldFocus = Ember.run.schedule 'afterRender', this, ->
+      if (@get('_state') or @get('state')) is 'inDOM'
+        @$('.ember-select-search .ember-text-field:eq(0)').focus()
```

Comment:
Kind of nitpicky, but jQuery has a `:first` selector that you can use instead of `:eq(0)`